lib/inspec/resources/users.rb

```diff
@@ -314,6 +336,59 @@ def credentials
 
       @cred_cache = @user_provider.credentials(@username) unless @user_provider.nil?
     end
+
+    # helper method for has_authorized_key matcher
+    # checks for key is present as part of authorized keys on the server
+    def authorized_key(key)
```

Comment:
shouldn't it be authorized_key?(key) as it returns true and false if I am not wrong here?

Reply:
changed `authorized_key(key)` to `get_authorized_keys` which would return the key/keys from the authorized_keys path.
Then the test for key is valid or not is done in `has_authorized_key?` method